### DISCUSSION TITLE
How to get user ip address?

### DISCUSSION THREAD

**Discussion by SarloAkrobata** [fastapi/fastapi#3378] [Questions] [answered]
Hello all, 

is there any solution to get user ip address when fastapi is served behind nginx?

I have added this in location section on nginx domain conf, but I can not get real ip  inside app.(BTW in Flask is working fine)
```
        proxy_set_header   Host                 $host;
        proxy_set_header   X-Real-IP            $remote_addr;
        proxy_set_header   X-Forwarded-For      $proxy_add_x_forwarded_for;
        proxy_set_header   X-Forwarded-Proto    $scheme;
```

**Comment by r-bar** [comment, marked as answer]
Using the `Header` dependency should let you access the `X-Real-IP` header.

```python
from fastapi import FastAPI, Depends, Header

app = FastAPI()

@app.get('/')
def index(real_ip: str = Header(None, alias='X-Real-IP')):
    return real_ip
```
Now if you start the server (in this case on port 8000) and hit it with a request with that `X-Real-IP` header set you should see it echo back.
```
❯ http :8000/ X-Real-IP:123.456.789.123
HTTP/1.1 200 OK
content-length: 17
content-type: application/json
date: Thu, 24 Jun 2021 06:36:44 GMT
server: uvicorn

"123.456.789.123"
```

**Reply by SarloAkrobata** [reply]
Thank you @r-bar. This works!

**Reply by ghandic** [reply]
Great explaination @r-bar ! @SarloAkrobata could you mark this as answered to help others looking for the solution?
